### PR TITLE
Use A-star for moving

### PR DESCRIPTION
**Pull request by spydon**
# Description

 - Use A-star for moving
 - Simple loading screen

### CODE REVIEW

File: lib/units/unit.dart [hunk 1]
```diff
@@ -222,49 +228,35 @@ abstract class Unit extends SpriteAnimationGroupComponent<UnitAnimationState>
...
-    for (var x = minX; x <= maxX; x++) {
-      for (var y = minY; y <= maxY; y++) {
-        blocks.add(Block(x, y));
-      }
-    }
-    return blocks;
+    final currentBlock = this.block;
+    final startingOffset = Offset(
+      currentBlock.x.toDouble(),
+      currentBlock.y.toDouble(),
```
Comment (luanpotter) [inline lib/units/unit.dart]: block.toVector2().toOffset()

File: lib/units/unit.dart [hunk 2]
```diff
@@ -222,49 +228,35 @@ abstract class Unit extends SpriteAnimationGroupComponent<UnitAnimationState>
...
+    occupiedBlocks.remove(reservedBlock);
+    target = reservedBlock = map.findCloseValidBlock(block);
+
+    path.clear();
+    path.addAll(
+      AStar(
+        rows: map.matrix.length,
+        columns: map.matrix.first.length,
+        start: startingOffset,
+        end: Offset(target!.x.toDouble(), target!.y.toDouble()),
```
Comment (luanpotter) [inline lib/units/unit.dart]: toVector2().toOffset()

Comment (luanpotter) [reply lib/units/unit.dart]: and below!

**Review by luanpotter — APPROVED** [review]
LGTM

File: lib/units/unit.dart (hunk 1 above)
Comment (spydon) [reply lib/units/unit.dart]: This method is already too expensive, we could add an extension to block that gives an offset.

Comment (spydon) [reply lib/units/unit.dart]: Right this is not in the map, so it should be fine, but doesn't really add readability to first convert it to a vector.

**Pull request merged**